File: api/internal/domain/models/game_attendance.go
package models

import "time"

type GameAttendance struct {
	UserID         string           `json:"userId"`
	UserName       string           `json:"userName"` // Added for display
	CheckedIn      bool             `json:"checkedIn"`
	Status         AttendanceStatus `json:"status"`
	CheckInComment string           `json:"checkInComment"`
	CreatedAt      time.Time        `json:"createdAt"`
	UpdatedAt      time.Time        `json:"updatedAt"`
}

File: bouncy-ui/lib/models/game_attendance.dart
class GameAttendance {
  final String userId;
  final bool checkedIn;
  final String checkInComment;
  final DateTime createdAt;
  final DateTime updatedAt;

  GameAttendance({
    required this.userId,
    required this.checkedIn,
    required this.checkInComment,
    required this.createdAt,
    required this.updatedAt,
  });

  factory GameAttendance.fromMap(Map<String, dynamic> map) {
    return GameAttendance(
      userId: map['UserID'] ?? map['userId'] ?? map['user_id'] ?? '',
      checkedIn: map['CheckedIn'] ?? map['checkedIn'] ?? false,
      checkInComment: map['CheckInComment'] ??
          map['checkInComment'] ??
          map['check_in_comment'] ??
          '',
      createdAt:
          DateTime.tryParse(map['CreatedAt'] ?? map['createdAt'] ?? '') ??
              DateTime.now(),
      updatedAt:
          DateTime.tryParse(map['UpdatedAt'] ?? map['updatedAt'] ?? '') ??
              DateTime.now(),
    );
  }
}
